### PR TITLE
Refactor parts of Project.swift and Xcode.swift

### PR DESCRIPTION
**Pull request by mdiep**
I did this on the plane the other day. ☺️

### CODE REVIEW

File: Source/CarthageKit/Project.swift [hunk 1]
```diff
@@ -176,7 +176,7 @@ public final class Project {
 		let cartfileURL = directoryURL.URLByAppendingPathComponent(CarthageProjectCartfilePath, isDirectory: false)
 		let privateCartfileURL = directoryURL.URLByAppendingPathComponent(CarthageProjectPrivateCartfilePath, isDirectory: false)
 
-		let isNoSuchFileError = { (error: CarthageError) -> Bool in
+		func isNoSuchFileError(error: CarthageError) -> Bool {
```
Comment (NachoSoto) [inline Source/CarthageKit/Project.swift]: Much better.

**Comment by NachoSoto** [comment]
Awesome 🌟

**Comment by NachoSoto** [comment]
you should simplify

File: Source/CarthageKit/Project.swift [hunk 2]
```diff
@@ -189,31 +189,20 @@ public final class Project {
 				return false
 			}
 		}
-
-		let cartfile = SignalProducer.attempt {
-				return Cartfile.fromFile(cartfileURL)
-			}
-			.flatMapError { error -> SignalProducer<Cartfile, CarthageError> in
-				if isNoSuchFileError(error) && NSFileManager.defaultManager().fileExistsAtPath(privateCartfileURL.path!) {
```
Comment (ikesyo) [inline Source/CarthageKit/Project.swift]: Loading `Cartfile` should be error out if both `Cartfile` and `Cartfile.private` dont't exist, and the error is successfully ignored in [`buildProjectInDirectoryURL` if `--no-skip-current` option is specified](https://github.com/Carthage/Carthage/blob/0.17/Source/carthage/Build.swift#L143-L149).

Thus removing the latter condition breaks the option appeared in #1366.

**Pull request merged**